src/evaluators.js

```diff
@@ -67,7 +67,22 @@ function createScopedEvaluatorFactory(unsafeRec, constants) {
   `);
 }
 
-export function createSafeEvaluatorFactory(unsafeRec, safeGlobal) {
+function rejectOrTransform(s, options) {
+  // control this with a root realm option.
+  const { infixBangResolver } = Object(options);
+  if (!infixBangResolver) {
+    // Just reject import expressions.
+    rejectImportExpressions(s);
+    return s;
+  }
+
+  // Do a full parse and regeneration.
+  const ast = parseSource(s, { infixBangResolver });
+  rejectImportExpressions(s, ast);
+  return generateSource(s, ast);
```

Comment:
Does this pattern protect us from inaccurate parsing allowing a import expression to slip through? Is the argument that we only rely on parse accuracy when we are also regenerating from ast, so an incorrectly parsed source is not passed through as is, where it still might be parsed differently by the target platform?